Engine/Results/LiveTradingResultHandler.cs

```diff
@@ -896,17 +901,15 @@ public void OrderEvent(OrderEvent newEvent)
         /// </summary>
         public void Exit()
         {
-            // If the algorithm was not successfully initialized, be sure to store the logs
-            // Update() will be unable to store the logs if the algorithm never full initialized
-            if (!_exitTriggered && _algorithm != null && !_algorithm.GetLocked())
+            if (!_exitTriggered)
             {
+                _exitTriggered = true;
                 ProcessSynchronousEvents(true);
-                StoreLog(_logStore);
+                lock (_logStoreLock)
+                {
+                    StoreLog(_logStore);
+                }
             }
-
-            _exitTriggered = true;
-
-            Update();
```

Comment:
note: justification here is that the engine is already calling `StoreFinalResult`, so running a final update was storing the result again -- are there concerns regarding the processing of any remaining messages in the queue?

Reply:
Here `ProcessSynchronousEvents(true)` will fetch any remaining log and we will store it.
Regarding the `Messages` queue, `Update()` thread will actually loop at https://github.com/QuantConnect/Lean/blob/master/Engine/Results/LiveTradingResultHandler.cs#L136 until he finishes processing all messages.